torch/testing/_internal/dist_utils.py

```diff
@@ -87,17 +87,25 @@ def new_test_method(self, *arg, **kwargs):
 def noop():
     pass
 
-def wait_until_node_failure(rank):
+def wait_until_node_failure(rank, sleep_duration=0.5, backoff=1):
     '''
     Loops until an RPC to the given rank fails. This is used to
     indicate that the node has failed in unit tests.
+
+    Args:
+        rank - Rank for which we are waiting to fail
+        sleep_duration - Amount to sleep in between checking if rank has failed
+        backoff - multiplier for sleep_duration
     '''
     while True:
         try:
+            time.sleep(sleep_duration)
```

Comment:
nit: Usually better to sleep after the operation rather than sleeping the first time we enter the loop.

Reply:
I changed this to sleep first otherwise for the UT I added, we'll send an RPC to node 0 which we're waiting for to timeout. As a result the timer in the ProcessGroup will be reset and we'll be forced to do a second sleep, so this approach allows us to save ~3s on this unit test at a minor cost (0.1 sleep) to other unit tests that use this function.